server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala

```diff
@@ -19,40 +19,26 @@
 package org.apache.james.jmap.rfc8621.contract
 
 import java.nio.charset.StandardCharsets
+import java.util.Base64
 
 import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
-import io.restassured.RestAssured
-import io.restassured.builder.RequestSpecBuilder
-import io.restassured.config.EncoderConfig.encoderConfig
-import io.restassured.config.RestAssuredConfig.newConfig
-import io.restassured.http.ContentType
+import io.restassured.RestAssured._
+import io.restassured.authentication.PreemptiveBasicAuthScheme
+import io.restassured.http.{ContentType, Header, Headers}
 import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
 import org.apache.http.HttpStatus
 import org.apache.james.GuiceJamesServer
-import org.apache.james.jmap.JMAPUrls.JMAP
-import org.apache.james.jmap.draft.JmapGuiceProbe
 import org.apache.james.jmap.rfc8621.contract.EchoMethodContract._
+import org.apache.james.jmap.rfc8621.contract.Fixture._
 import org.apache.james.jmap.rfc8621.contract.tags.CategoryTags
+import org.apache.james.utils.DataProbeImpl
 import org.junit.jupiter.api.{BeforeEach, Tag, Test}
 
 object EchoMethodContract {
+  private val authScheme: PreemptiveBasicAuthScheme = new PreemptiveBasicAuthScheme
```

Comment:
I would suggest adding in the Fixture:

```
def authSheme(user: Username, pass: String): PreemptiveBasicAuthScheme = ???
```

So that here we just write:

```
requestSpecification = baseRequestSpecBuilder(server)
        .getValue)
        .setAuth(authScheme(BOB, BOB_PASSWORD))
```

Reply:
we should accept `List<UserCredential>` because we have a case where were need more than 1 user.

Reply:
IMO no because a given request is done by a signgle user.

We can very well override the default user in a specific test by writting:

```
given()
    .auth(authSheme(ALICE, ALICE_PASSWORD))
   ...
```

When needed.

Reply:
OK I agree with it